### PR TITLE
Clear social invitations when adding a user

### PR DESCRIPTION
**Pull request by zapu**
In a rare case where user is invited via social assertion that is not automatically resolved, clear the old assertions when someone manually adds user as member.

PR is still work in progress, we are waiting for some code in https://github.com/keybase/client/pull/8414 to simplify this.

### CODE REVIEW

**Comment by mlsteele** [comment]
If a team had a ton of invites then this could take a long time. How about giving `tryToCompleteInvites` a `ctx` which is time limited using `context.WithTimeout` of, say, 2 seconds. Just so that this has no chance of hanging someone's add-user intention.

**Comment by zapu** [comment]
Good idea, I will try that, thanks!

**Comment by zapu** [comment]
@mlsteele would something like this work for optimized proof checking? I'll refactor and clean the code up though

File: go/teams/service_helper.go [hunk 1]
```diff
@@ -120,6 +121,98 @@ func SetRoleReader(ctx context.Context, g *libkb.GlobalContext, teamname, userna
...
+		g.Log.CDebugf(ctx, "Resolve result is: %+v", resolveResult)
+		if resolveResult.GetError() != nil || resolveResult.GetUID() != uv.Uid {
+			// Cannot resolve invitation or it does not match user
+			continue
+		}
+
+		assertion := fmt.Sprintf("%s@%s+uid:%s", string(invite.Name), ityp, uv.Uid)
+		g.Log.CDebugf(ctx, "Assertion is: %s", assertion)
+
+		arg := keybase1.Identify2Arg{
```
Comment (mlsteele) [inline go/teams/service_helper.go]: Alright how about rebasing and using `verifyResolveResult` to validate the resolution. Would that fit?

File: go/teams/service_helper.go [hunk 2]
```diff
@@ -156,7 +249,9 @@ func AddMember(ctx context.Context, g *libkb.GlobalContext, teamname, username s
 		}
 		req.None = []keybase1.UserVersion{existingUV}
 	}
-
+	if err := tryToCompleteInvites(ctx, g, t, username, uv, &req); err != nil {
```
Comment (mlsteele) [inline go/teams/service_helper.go]: pass in a time limited context here?

File: go/teams/service_helper.go (hunk 1 above)
Comment (zapu) [reply go/teams/service_helper.go]: Tried to use new API, but I have a feeling I missed few things and it's still needlessly complicated. I'd appreciate if you can take a look. Thanks!

**Pull request merged**